File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int count_0=0, count_1=0, count_2=0;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]==0)
                count_0++;
            else if(nums[i]==1)
                count_1++;
            else
                count_2++;
        }
        
        // cout<<count_0<<count_1<<count_2<<endl;
        for(int i=0;i<count_0;i++)
            nums[i]=0;
        
        
        for(int i=count_0;i<count_1+count_0;i++)
            nums[i]=1;
        
        for(int i=count_1+count_0;i<count_2+count_1+count_0;i++)
            nums[i]=2;
    }
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        low=0;
        mid=0;
        high=len(nums)-1;
        for i in range(mid,len(nums)):
            if(mid > high):
                break
            if(nums[mid] == 0):
                nums[low],nums[mid] = nums[mid],nums[low]
                mid+=1
                low+=1
                # print("first block")
                # print("Iteration",i,nums)
                # print("low:",low," mid:",mid," high:",high," i:",i," nums[i]:",nums[i])
            elif(nums[mid] == 1):
                mid+=1
                # print("second block")
                # print("Iteration",i,nums)
                # print("low:",low," mid:",mid," high:",high," i:",i," nums[i]:",nums[i])


            elif(nums[mid] == 2):
                nums[high],nums[mid] = nums[mid],nums[high]
                high-=1
                # print("third block")
                # print("Iteration",i,nums)
                # print("low:",low," mid:",mid," high:",high," i:",i," nums[i]:",nums[i])
